pgmpy/tests/test_prediction/test_NaiveAdjustmentRegressor.py

```diff
@@ -0,0 +1,407 @@
+    dag = dag.with_role("adjustment", [])
+    return NaiveAdjustmentRegressor(causal_graph=dag, estimator=LinearRegression())
+
+
+@parametrize_with_checks([make_estimator()])
+def test_sklearn_compatibility(estimator, check):
+    """Test sklearn compatibility using parametrize_with_checks."""
+    check(estimator)
+
+
+def test_basic_functionality_with_adjustment():
+    """Test basic fit and predict functionality with synthetic causal data."""
+
+    lgbn = DAG.from_dagitty(
+        "dag { Z -> X [beta=0.5] X -> Y [beta=2.0] Z -> Y [beta=1.5] }"
+    )
+
+    data = lgbn.simulate(1000, seed=42)
+
+    dag = DAG(
+        ebunch=[("Z", "X"), ("Z", "Y"), ("X", "Y")],
+        roles={"exposure": "X", "outcome": "Y", "adjustment": ["Z"]},
+    )
+
+    estimators = [
+        LinearRegression(),
+        RandomForestRegressor(n_estimators=10, random_state=42),
+        DummyRegressor(),
+    ]
+
+    for base_est in estimators:
+        regressor = NaiveAdjustmentRegressor(causal_graph=dag, estimator=base_est)
+
+        train_size = int(0.7 * len(data))
+        X_train = data[["X", "Z"]].iloc[:train_size]
+        y_train = data["Y"].iloc[:train_size]
+        X_test = data[["X", "Z"]].iloc[train_size:]
+        y_test = data["Y"].iloc[train_size:]
+
+        regressor.fit(X_train, y_train)
+        predictions = regressor.predict(X_test)
+
+        assert len(predictions) == len(y_test)
+        assert regressor.exposure_var_ == "X"
+        assert regressor.outcome_var_ == "Y"
+        assert regressor.adjustment_vars_ == ["Z"]
+
+        if isinstance(base_est, LinearRegression):
+            # Expected coefficients: [X_coef=2.0, Z_coef=1.5] from simulation
+            # Feature order is [X, Z] (exposure + adjustment)
+            expected_coefs = [2.0, 1.5]  # beta coefficients from DAG simulation
+            actual_coefs = regressor.estimator_.coef_
+
+            # Check coefficients are close to expected values (allowing for noise)
+            np.testing.assert_allclose(actual_coefs, expected_coefs, atol=0.2)
+
+            # Check intercept is close to 0 (no baseline effect in simulation)
+            assert abs(regressor.estimator_.intercept_) < 0.1
+
+        feature_names = regressor.get_feature_names_out()
+        expected_features = ["X", "Z"]
+        assert list(feature_names) == expected_features
+
+
+def test_dataframe_input_for_both_x_and_y():
+    """Test that regressor works when both X and y are DataFrames."""
+    lgbn = DAG.from_dagitty(
+        "dag { Z -> X [beta=0.5] X -> Y [beta=2.0] Z -> Y [beta=1.5] }"
+    )
+
+    data = lgbn.simulate(100, seed=42)
+
+    dag = DAG(
+        ebunch=[("Z", "X"), ("Z", "Y"), ("X", "Y")],
+        roles={"exposure": "X", "outcome": "Y", "adjustment": ["Z"]},
+    )
+
+    regressor = NaiveAdjustmentRegressor(causal_graph=dag)
+
+    X_df = data[["X", "Z"]]
+    y_df = data[["Y"]]
+
+    regressor.fit(X_df, y_df)
+    predictions = regressor.predict(X_df)
```

Comment:
I think we should be testing the predicted values here. This shouldn't be very different from the true values.